### PR TITLE
Support .NET Core

### LINKED ISSUE #203: Using FsUnit for more expressive tests
At the moment, the exercises used NUnit's `Assert` class' methods to verify behavior. While this works, it's not really a nice functional API. Perhaps it would make sense to use [FsUnit](http://fsprojects.github.io/FsUnit/), which provides a functional wrapper around the tests.

As an example, this is how an assertion looks like today:

``` fsharp
Assert.That(1, Is.EqualTo(1))
```

And this is how it would look with FsUnit:

``` fsharp
1 |> should equal 1
```

I much prefer the second option. It would require people to install an additional package before the tests can be run, but that won't really be a problem as I'm working on creating a pre-defined project for each exercise, which would already contain the required libraries.

cc @jwood803 @rmunn @robkeim @rebelwarrior @pminten

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This PR fixes several issues:

- Create a project for each exercise. This makes getting started a lot simpler. (#309)
- All the exercises now have a default, stub file. This also makes getting started easier. All the easy (difficulty 1-3) exercises have stub methods with fully specified types, also to help people get started. The medium (difficulty 4-7) exercises have stub methods without types (type-inference) and the hard exercises (difficulty 8-10) only have an empty stub module. (#229)
- Allow exercises to run on .NET Core. This is great for our cross-platform support. Users on Mac or Unix should now have the same options (and command-line interface) as Windows users. (#308)
- FsUnit is used to make the tests appear more functional. (#203)

What is left to do is:
- Replace `[<TestCase>]` tests with individual tests. (#298)
- Replace NUnit with Xunit. This should be easier now that we've moved to FsUnit. (#297)
- Fix Travis build.

Note: it is still a work-in-progress, but if you'd be willing to take a look, that would be great!

CC: @jpreese Would you also mind taking a look?

This fixes #203, #229, #251, #277, #297, #308, #309

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                System.IO.File.WriteAllText(testFile.FullPath, updatedContents);
+            }
+        }
+    });
+
+Task("TestRefactoringProjects")
+    .IsDependentOn("EnableAllTests")
+    .Does(() => {
+        var refactoringSolution = ParseSolution(refactoringSolutionPath);
+        Parallel.ForEach(refactoringSolution.Projects, parallelOptions, (project) => DotNetCoreTest(project.Path.FullPath));
```
Comment (robkeim) [inline build.cake]: Do you need the `noBuild` configuration setting like in the C# track?
https://github.com/exercism/csharp/pull/387/files

File: exercises/Exercises.All.sln [hunk 2]
```diff
@@ -0,0 +1,1502 @@
...
+Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Wordy", "wordy\Wordy.fsproj", "{B14FF411-3EDE-4E2F-98AB-78CBDFFBAF48}"
+EndProject
+Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "ZebraPuzzle", "zebra-puzzle\ZebraPuzzle.fsproj", "{3C57692F-7E65-4C6F-9F2D-3962A9007EE2}"
+EndProject
+Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Zipper", "zipper\Zipper.fsproj", "{1DEE9AFA-D88A-4143-B53D-E494FF331FC8}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Debug|x64 = Debug|x64
```
Comment (robkeim) [inline exercises/Exercises.All.sln]: Minor: we don't need `x64` or `x86` builds the `Any CPU` is enough

File: exercises/acronym/AcronymTest.fs [hunk 3]
```diff
@@ -1,64 +1,65 @@
 module AcronymTest
 
 open NUnit.Framework
+open FsUnit
 open Acronym
 
 [<Test>]
 let ``Empty string abbreviated to empty string`` () =
-    Assert.That(acronym "", Is.EqualTo(""))
+    acronym "" |> should equal ""
```
Comment (robkeim) [inline exercises/acronym/AcronymTest.fs]: `should be NullOrEmptyString`

File: generate.fsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
+open System
```
Comment (robkeim) [inline generate.fsx]: Should this be checked in? It seems like a tool that you are using for the migration but it won't be useful after that.

Comment (ErikSchierboom) [reply generate.fsx]: It should indeed be removed. I was planning to wait for this, but as I don't really need it anymore, I'll remove it now.

File: exercises/acronym/AcronymTest.fs (hunk 3 above)
Comment (ErikSchierboom) [reply exercises/acronym/AcronymTest.fs]: Well, I don't think it should actually. The test case states explicitly that it should be abbreviated to an empty string. If I were to use `NullOrEmptyString`, returning `null` would also pass the test.

File: exercises/Exercises.All.sln (hunk 2 above)
Comment (ErikSchierboom) [reply exercises/Exercises.All.sln]: Good point!

File: build.cake (hunk 1 above)
Comment (ErikSchierboom) [reply build.cake]: No I don't, as the .NET CLI version 2.0 will automatically restore packages when the `dotnet test` command runs. In the C# track, we use the older version of the CLI which requires an explicit `dotnet restore` before that.

File: exercises/allergies/AllergiesTest.fs [hunk 5]
```diff
@@ -1,74 +1,79 @@
...
-    Assert.That(allergicTo Allergen.Strawberries 0, Is.False)
+    allergicTo Allergen.Peanuts 0 |> should equal false
+    allergicTo Allergen.Cats 0 |> should equal false
+    allergicTo Allergen.Strawberries 0 |> should equal false
 
 [<Test>]
 [<Ignore("Remove to run test")>]
 let ``Allergic to eggs`` () =
-    Assert.That(allergicTo Allergen.Eggs 1, Is.True)
+    allergicTo Allergen.Eggs 1 |> should equal true
```
Comment (ErikSchierboom) [inline exercises/allergies/AllergiesTest.fs]: @robkeim What do you think about using `should equal true`? The docs also mention `should be True`, but I was hesistant to use that option as the uppercase T in `True` could throw newcomers off I think,

File: exercises/acronym/AcronymTest.fs (hunk 3 above)
Comment (ErikSchierboom) [reply exercises/acronym/AcronymTest.fs]: Note: there also is a `EmptyString` variant, which I'll use.

File: build.cake (hunk 1 above)
Comment (robkeim) [reply build.cake]: Ah nice!

File: exercises/acronym/AcronymTest.fs (hunk 3 above)
Comment (robkeim) [reply exercises/acronym/AcronymTest.fs]: Sounds good!

File: generate.fsx (hunk 4 above)
Comment (robkeim) [reply generate.fsx]: Ok!

File: exercises/allergies/AllergiesTest.fs (hunk 5 above)
Comment (robkeim) [reply exercises/allergies/AllergiesTest.fs]: Given there's already a mix of `should be` and `should equal` I'm fine going with `should equal true`.

**Comment by ErikSchierboom** [comment]
@robkeim The PR has been updated and I think it is now correctly working. Could you perhaps review the PR again? If you deem it correct, feel free to merge.

File: docs/TESTS.md [hunk 6]
```diff
@@ -1,34 +1,21 @@
...
-Click on "New Project" and you should see an entry for Visual F# and click on "Class Library".
+```bash
+dotnet test
+```
 
-![New Project](http://x.exercism.io/v3/tracks/fsharp/docs/img/newProject.png)
+## Solving the exercise
 
-Install [NUnit](http://nunit.org/index.php?p=download) via NuGet package manager with the NUnit.Console package. This will add the reference to the project.
+Solving an exercise means making all its tests pass. By default, only one test (the first one) is executed when you run the tests. This is intentional, as it allows you to focus on just making that one test pass. Once it passes, you can enable the next test by removing `Skip = "Remove to run test"` from the test's `[<Fact>]` or `[<Theory?]` attribute. When all tests have been enabled and your implementation makes them all pass, you'll have solved the exercise!
```
Comment (robkeim) [inline docs/TESTS.md]: Minor typo: `[<Theory>]`

File: exercises/acronym/AcronymTest.fs (hunk 3 above)
Comment (robkeim) [reply exercises/acronym/AcronymTest.fs]: @ErikSchierboom did you decide not to update this to `EmptyString` for some reason?

File: docs/TESTS.md (hunk 6 above)
Comment (robkeim) [reply docs/TESTS.md]: Fixed!

**Comment by ErikSchierboom** [comment]
@robkeim I forgot about the EmptyString. Perhaps I can fix it later in a new commit?

**Comment by robkeim** [comment]
@ErikSchierboom yes I figured you could do it in another PR. Once the fix of my typo is green I'll go ahead and merge this branch.

**Comment by robkeim** [comment]
Thanks for your hard work on this @ErikSchierboom! Great job 🎉 🎉 🎉

**Comment by ErikSchierboom** [comment]
Hurray!

**Comment by ErikSchierboom** [comment]
image

@Mousaka @kytrinyx The F# track has now been updated to support .NET Core, which is its biggest overhaul yet!

**Comment by kytrinyx** [comment]
@ErikSchierboom Fantastic. Thanks so much for taking the time to do this overhaul... this is huge.

**Pull request merged**